proxy/fadein_test.go

```diff
@@ -0,0 +1,253 @@
+package proxy_test
+
+import "testing"
+
+/*
+The following tests address the fade-in behavior expectations as observed from the perspective of the backend
+endpoints of a single route.
+
+Available data:
+- fade-in configuration: this contains the fade-in duration and optionally the fade-in degree.
+- detection time: this is the time when Skipper detected an endpoint that it didn't know about.
+- created time, optional: this is external information about the created of an endpoint, when available. E.g.
+  it can be the time when a Kubernetes pod became ready. It can be used as a fallback value for the detection
+  time in cases when Skipper (re)starts. When present, it renders any detection time past this value as invalid.
+- synced detection time, optional: it's the detection time of an endpoint shared across multiple Skipper
+  instances, stored in Redis or shared as a Skipper swarm datagram. It can be used to continue the fade-in when
+  Skipper gets (re)started. It has a precedence over the external created time in these cases.
```

Comment:
If we would store it in swarm/redis we also need the case when it's not present, because of errors of the redis/swarm.

Reply:
isn't that covered by the combination of state and event: proxies without syncing + proxy started during fade-in?